casper/src/main/scala/coop/rchain/casper/SignDeployment.scala

```diff
@@ -28,10 +30,10 @@ object SignDeployment {
   }
 
   def verify(deployData: DeployData): Option[Boolean] =
-    SignaturesAlg(deployData.sigAlgorithm).map { alg =>
+    SignaturesAlg(deployData.sigAlgorithm) >>= { alg =>
       val toVerify = clear(deployData).toByteString.toByteArray
       val hash     = Blake2b256.hash(toVerify)
-      alg.verify(hash, deployData.sig.toByteArray, PublicKey(deployData.deployer.toByteArray))
+      Try(alg.verify(hash, deployData.sig.toByteArray, PublicKey(deployData.deployer.toByteArray))).toOption
```

Comment:
It doesn't make sense that alg.verify throws even if the underlying implementation does throw. 

Ed25519 implementation already has a try/catch around the code that could fail.

So, instead of adding a Try here, we should add a try in the implementation of Secp256k1

Reply:
@odeac you are not correct, it was actuallt `Ed25519` that was throwing erros! That's why the Try was added in the first place!